st2common/st2common/garbage_collection/inquiries.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_config import cfg
+
+from st2actions.container.base import get_runner_container
+from st2common.constants import action as action_constants
+from st2common.models.db.auth import UserDB
+from st2common.persistence.execution import ActionExecution
+from st2common.services import action as action_service
+from st2common.services import executions
+from st2common.util import action_db as action_utils
+from st2common.util.action_db import (get_action_by_ref, get_runnertype_by_name)
+from st2common.util.date import get_datetime_utc_now
+
+__all__ = [
+    'purge_inquiries',
+]
+
+
+def purge_inquiries(logger):
+    """Purge Inquiries that have exceeded their configured TTL
+
+    At the moment, Inquiries do not have their own database model, so this function effectively
+    is another, more specialized GC for executions. It will look for executions with a 'pending'
+    status that use the 'inquirer' runner, which is the current definition for an Inquiry. Then
+    it will force-fail those that:
+
+    - Have a nonzero TTL
+    - Have existed longer than their TTL
+    """
+
+    # Get all existing Inquiries
+    filters = {'runner__name': 'inquirer', 'status': action_constants.LIVEACTION_STATUS_PENDING}
+    inquiries = list(ActionExecution.query(**filters))
+
+    gc_count = 0
+
+    # Inspect each Inquiry, and determine if TTL is expired
+    for inquiry in inquiries:
+
+        ttl = inquiry.result.get('ttl')
```

Comment:
Should this be `inquiry.result.get('ttl', some_default_ttl)`?

Reply:
I can do that, but it's a bit redundant, as the runner already enforces defaults and guarantees they're placed into `result`. Anything past that point I've been excluding a default value for `.get()` calls so that if the default needs to change, it's just in one place.